Sources/WalletConnectRelay/Dispatching.swift

```diff
@@ -84,28 +102,51 @@ final class Dispatcher: NSObject, Dispatching {
     }
 
     func connect() throws {
-        try socketConnectionHandler.handleConnect()
+        try socketConnectionHandler?.handleConnect()
     }
 
     func disconnect(closeCode: URLSessionWebSocketTask.CloseCode) throws {
-        try socketConnectionHandler.handleDisconnect(closeCode: closeCode)
+        try socketConnectionHandler?.handleDisconnect(closeCode: closeCode)
+    }
+    
+    private func setUpSocketConnection() {
+        setUpSocket()
+        setUpWebSocketSession()
+        setUpSocketConnectionObserving()
```

Comment:
same question, is it necessary to `setUpSocketConnectionObserving` on reconnection?